src/main/java/com/marklogic/kafka/connect/sink/DefaultSinkRecordConverter.java

```diff
@@ -31,18 +34,33 @@ public DefaultSinkRecordConverter(Map<String, String> kafkaConfig) {
 		if (val != null && val.trim().length() > 0) {
 			format = Format.valueOf(val.toUpperCase());
 		}
-
 		val = kafkaConfig.get(MarkLogicSinkConfig.DOCUMENT_MIMETYPE);
 		if (val != null && val.trim().length() > 0) {
 			mimeType = val;
 		}
 	}
-
+	
 	@Override
 	public DocumentWriteOperation convert(SinkRecord sinkRecord) {
-		return documentWriteOperationBuilder.build(toContent(sinkRecord));
+		return documentWriteOperationBuilder.build(toContent(sinkRecord), addTopicToCollections(sinkRecord.topic(), addTopicToCollections ) );
 	}
 
+	/**
+	 * 
+	 *
+	 * @param topic, addTopicToCollections
+	 * @return  metadata 
+	 */
+	public DocumentMetadataHandle addTopicToCollections (String topic, Boolean addTopicToCollections) {
```

Comment:
For open source, we have to be careful about what methods are public. As part of semantic versioning, we shouldn't be changing public methods in between minor or bugfix releases. But we can fiddle with protected/private methods without worrying about backwards compatibility. So this should be protected or private - I'd do protected since the other non-public method in this class is protected.